iot_config/src/helium_netids.rs

```diff
@@ -0,0 +1,377 @@
+        )
+    }
+
+    async fn claim_addrs(&mut self, new_addrs: &[u32]) -> Result<(), Self::Error> {
+        let mut query_builder: sqlx::QueryBuilder<sqlx::Postgres> =
+            sqlx::QueryBuilder::new(" insert into helium_used_devaddrs (devaddr) ");
+        query_builder.push_values(new_addrs, |mut builder, addr| {
+            builder.push_bind(*addr as i32);
+        });
+        Ok(query_builder.build().execute(&*self).await.map(|_| ())?)
+    }
+
+    async fn release_addrs(&mut self, released_addrs: &[u32]) -> Result<(), Self::Error> {
+        let mut query_builder: sqlx::QueryBuilder<sqlx::Postgres> =
+            sqlx::QueryBuilder::new(" delete from helium_used_devaddrs where devaddr in ");
+        query_builder.push_values(released_addrs, |mut builder, addr| {
+            builder.push_bind(*addr as i32);
+        });
+        Ok(query_builder.build().execute(&*self).await.map(|_| ())?)
+    }
+}
+
+#[async_trait::async_trait]
+impl AddressStore for sqlx::Transaction<'_, sqlx::Postgres> {
+    type Error = sqlx::Error;
+
+    async fn get_used_addrs(&mut self) -> Result<Vec<u32>, Self::Error> {
+        Ok(sqlx::query_scalar::<_, i32>(
+            " select devaddr from helium_used_devaddrs order by devaddr asc ",
+        )
+        .fetch_all(self)
+        .await?
+        .into_iter()
+        .map(|addr| addr as u32)
+        .collect::<Vec<u32>>())
+    }
+
+    async fn claim_addrs(&mut self, new_addrs: &[u32]) -> Result<(), Self::Error> {
+        let mut query_builder: sqlx::QueryBuilder<sqlx::Postgres> =
+            sqlx::QueryBuilder::new(" insert into helium_used_devaddrs (devaddr) ");
+        query_builder.push_values(new_addrs, |mut builder, addr| {
+            builder.push_bind(*addr as i32);
+        });
+        Ok(query_builder.build().execute(self).await.map(|_| ())?)
+    }
+
+    async fn release_addrs(&mut self, released_addrs: &[u32]) -> Result<(), Self::Error> {
+        let mut query_builder: sqlx::QueryBuilder<sqlx::Postgres> =
+            sqlx::QueryBuilder::new(" delete from helium_used_devaddrs where devaddr in ");
+        query_builder.push_values(released_addrs, |mut builder, addr| {
+            builder.push_bind(*addr as i32);
+        });
+        Ok(query_builder.build().execute(self).await.map(|_| ())?)
+    }
+}
+
+pub async fn checkout_devaddr_constraints<S>(
+    addr_store: &mut S,
+    count: u64,
+    net_id: HeliumNetId,
+) -> Result<Vec<DevAddrConstraint>, DevAddrConstraintsError<S::Error>>
+where
+    S: AddressStore,
+{
+    let addr_range = net_id.addr_range();
+    let used_addrs = addr_store
+        .get_used_addrs()
+        .await
+        .map_err(DevAddrConstraintsError::AddressStore)?;
+    let range_start = *addr_range.start();
+    let range_end = *addr_range.end();
+    let last_used = if let Some(last) = used_addrs.last() {
+        *last
+    } else {
+        range_start
+    };
+    let used_range = (range_start..=last_used).collect::<HashSet<u32>>();
```

Comment:
`used_addrs` is not Net ID aware. 
If you attempt to checkout devaddrs in a Type0 NetID after an existing Type6, this function will never finish.

```rust
    #[tokio::test]
    async fn allocate_across_net_id() {
        let mut used_addrs = vec![];
        checkout_devaddr_constraints(&mut used_addrs, 8, HeliumNetId::Type6)
            .await
            .expect("testing allocation");
        checkout_devaddr_constraints(&mut used_addrs, 8, HeliumNetId::Type3)
            .await
            .expect("special request allocation");
        checkout_devaddr_constraints(&mut used_addrs, 8, HeliumNetId::Type0)
            .await
            .expect("average allocation");

        assert_eq!(8 + 8 + 8, used_addrs.len());
    }
```

Reply:
fixed: [56ba443](https://github.com/helium/oracles/pull/514/commits/56ba4431b2ce7adcc0d2bb52d98b9fe80cfb3936)